### ISSUE TITLE
PyCharm IDE inspection warnings

### ISSUE THREAD

**Issue by fleXible** [custom-components/pyscript#17]
Since the refactoring away from instancing `State`, `Handler` and the other classes, my PyCharm IDE warns about two inspections, and I think it's right to do so:
- `Usually first parameter of a method is named 'self'`
To fix this, the `@staticmethod`decorator should be used ([explanation](https://www.tutorialspoint.com/class-method-vs-static-method-in-python))
- `Unresolved attribute reference 'notify' for class 'State'`
Class variables are defined and initialized in class scope ([Python docs](https://docs.python.org/3/tutorial/classes.html#class-and-instance-variables)).
This would also do away with the ìnit() class method, as it is not needed anymore. As init, just set the `hass` class variable

**Comment by craigbarratt** [comment]
Thanks for the info - I'm actually relatively new to Python (started in April), so it's really helpful to learn about the right way to do things.  Let me read about this some more.

I've actually tried three different approaches to dealing with the state information maintained by State, Function, Event etc, and I'm not 100% happy with any of them, although the current approach seems best.  The first approach was using global variables inside each file, which `pylint` obviously complained about.  The second approach was to create one instance each of the State, Function, Event classes, and passing them to all the functions that needed it.  My third approach was to not instantiate those class, and use the class object itself as a container for the variables, which is the current version.

It looks like `@classmethod` decorators and adding a `cls` argument is the right approach.  I made this change to `state.py` and pushed this commit:  5665558.  Does that pass your IDE correctly now?  I still need to make the same changes to `Event`, `Function` and others...
